### ISSUE TITLE
[KotlinJs] Fatal exception in coroutines machinery for CancellableContinuation

### ISSUE THREAD

**Issue by mirland** [Kotlin/kotlinx.coroutines#2977]
I found an issue in KotlinJs. In Kotlin JVM and native (ios target) seems to be working as expected.

The exception message is: 
```
"Job StandaloneCoroutine{Completed}@1 is already complete or completing, but is being completed with CompletedExceptionally[JobCancellationException: StandaloneCoroutine has completed normally; job=StandaloneCoroutine{Completed}@1]"
```

Steps to reproduce it:
First of all, create this interface
```kotlin
interface Foo<in Param, Output> {
    suspend operator fun invoke(params: Param): Output =
        execute(params)

    suspend fun execute(params: Param): Output
}
```

If you define an instance of foo as:
```kotlin
class Foo1 : Foo<Unit, Boolean> {
    override suspend fun execute(params: Unit): Boolean {
        delay(1500)
        return true
    }
}
```
Then in the main app, you call
```kotlin
fun main() {
    GlobalScope.launch {
        val foo1 = Foo1()
        println("Foo1: ${foo1.invoke(Unit)}")
    }
}
```
It generates a good output: `Foo1: true`. However, if you define `Output` as `Unit` and you test the same, the app crashes.

```kotlin
class Foo2 : Foo<Unit, Unit> {
    override suspend fun execute(params: Unit): Unit {
        delay(1500)
    }
}


fun main() {
    GlobalScope.launch {
        val foo2 = Foo2()
        println("Foo2: ${foo2.invoke(Unit)}")
    }
}
```
Output:
```
Foo2: kotlin.Unit
CoroutineExceptionHandlerImpl.kt?f1f9:11 CoroutinesInternalError {message: "Fatal exception in coroutines machinery for Cancel…n' method and report this incident to maintainers", cause: IllegalStateException, name: 'CoroutinesInternalError', stack: 'CoroutinesInternalError: Fatal exception in corout…bpack-internal:///./kotlin/app-web.js:82470:9)'}
```

To sum up, you cannot define `Output` as `Unit` in `Foo`

Versions:
Compiler = IR
Kotlin version = "1.5.30"
Coroutines version = "1.5.2-native-mt" // "1.5.2" has the same issue
